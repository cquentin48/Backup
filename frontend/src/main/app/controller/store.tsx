import { configureStore } from "@reduxjs/toolkit";

import deviceReducer from "./deviceMainInfos/loadDeviceSlice";
import filterReducer from "./deviceMainInfos/filterSlice";
import snapshotReducer from "./deviceMainInfos/loadSnapshotSlice";

export const store = configureStore({
    reducer: {
        device: deviceReducer,
        filter: filterReducer,
        snapshot: snapshotReducer
    }
})

export type AppState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
