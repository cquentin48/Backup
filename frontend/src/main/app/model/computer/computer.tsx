import Snapshot from "./saves";

/**
 * Computer Data class
 */
class Computer {
    name: string;
    processor: string;
    cores: number;
    memory: number;
    operatingSystem: string;
    snapshots: [Snapshot?];

    /**
     * Constructor method
     * @param name computer network name
     * @param processor computer processor model
     * @param cores computer cores
     * @param memory computer volatile memory
     * @param operatingSystem computer operating system name
     * @param snapshots header of every save uploaded from the device
     */
    constructor (
        name: string, processor: string, cores: number, memory: number,
        operatingSystem: string, snapshots: [Snapshot?]) {
        this.name = name;
        this.processor = processor;
        this.cores = cores;
        this.memory = memory;
        this.operatingSystem = operatingSystem;
        this.snapshots = snapshots;
    }

    /**
     * Format memory for visualisation
     * @param memory volatile memory quantity
     * @returns string
     */
    formatBytes (memory: number): string {
        const units = ['Bytes', 'kB', 'MB', 'GB', 'TB']
        let formattedMemory: number = memory;
        let unitIndex: number = 0;
        while (formattedMemory > 1000) {
            formattedMemory /= 1000;
            unitIndex += 1;
        }
        formattedMemory = Math.round(formattedMemory * 100) / 100
        return `${formattedMemory} ${units[unitIndex]}`
    }
}

export default Computer;
