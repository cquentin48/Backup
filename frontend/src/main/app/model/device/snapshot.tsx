/**
 * Device snapshot header class
 * Composed of the index of database stored data and the upload date.
 */
export default class SnapshotID {
    id: string;
    uploadDate: Date;
    operatingSystem: string;

    /**
     * Construction method
     * @param { string } id snapshot database id
     * @param { string } uploadDate snapshot date upload (Must be of format ``YYYY``-``MM``-``DD``)
     */
    constructor (id: string, uploadDate: string, operatingSystem: string) {
        this.id = id;
        const uploadDateArray = uploadDate.split('-');
        this.uploadDate = new Date(
            parseInt(uploadDateArray[0]),
            parseInt(uploadDateArray[1]),
            parseInt(uploadDateArray[2])
        );
        this.operatingSystem = operatingSystem;
    }

    /**
     * Format the snapshot date for display in the web application according to the chosen language
     * @returns {string} Localized snapshot upload date
     */
    localizedDate (): string {
        return this.uploadDate.toLocaleDateString(window.navigator.language, {
            weekday: "long",
            year: "numeric",
            month: "long",
            day: "numeric"
        }
        )
    }
}
