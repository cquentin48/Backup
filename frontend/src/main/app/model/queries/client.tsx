import {
    ApolloClient, type ApolloQueryResult,
    type DocumentNode, HttpLink, InMemoryCache,
    type NormalizedCacheObject
} from '@apollo/client';
import fetch from 'cross-fetch';

import { Config } from '../../config/backupConfig';
import type BasicQueryParameters from './basicQueryParameters';

/**
 * Backup frontend graphQL query client interface manager
 */
class BackupQueryClient {
    private readonly query_client: ApolloClient<NormalizedCacheObject>;

    constructor () {
        this.query_client = new ApolloClient({
            uri: Config.GQL_BACKENDURI,
            cache: new InMemoryCache(),
            link: new HttpLink({ uri: Config.GQL_BACKENDURI, fetch })
        });
    }

    /**
     * Getter for the query client (used only in the app component)
     * @returns {ApolloClient<NormalizedCacheObject>} GraphQL query client
     */
    get_query_client (): ApolloClient<NormalizedCacheObject> {
        return this.query_client;
    }

    /**
     * Execute a graphQL query and returns the results
     * @param { string } query GraphQL query
     * @param { BasicQueryParameters } variables query variables
     * @returns { Promise<ApolloQueryResult<any>> } Result of the query
     */
    async execute_query (query: DocumentNode, variables: BasicQueryParameters | undefined): Promise<ApolloQueryResult<any>> {
        const data = await this.query_client.query({
            query,
            variables
        })
        return data
    }
}

const gqlClient = new BackupQueryClient();

export default gqlClient;
