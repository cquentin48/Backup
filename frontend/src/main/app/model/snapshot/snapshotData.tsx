import NotFoundError from "../exception/errors/notFoundError";
import AlreadyAddedWarning from "../exception/warning/alreadyAdded";
import { SnapshotSoftware } from "./snapshotLibrary";

/**
 * Snapshot data fetched from the query
 */
export class SnapshotData {
    /**
     * Every software
     */
    softwares: SnapshotSoftware[];

    /**
     * Class object constructor method
     */
    constructor () {
        this.softwares = [];
    }

    /**
     *
     * @param {string} softwareVersion Version of the software
     * @param {string} softwareName Software name
     * @param {string} softwareInstallType Software installation type (e.g. ``apt``, ``snap``)
     * @throws {AlreadyAddedWarning} Already added sofware in the array
     */
    addSoftware (softwareVersion: string, softwareName: string, softwareInstallType: string): void {
        const newSoftware = new SnapshotSoftware(
            softwareVersion,
            softwareName,
            softwareInstallType
        )
        if (this.softwares.filter((software) => {
            const sameInstallType = software.softwareInstallType === softwareInstallType
            const sameName = software.softwareName === softwareName
            const sameVersion = software.softwareVersion === softwareVersion
            return sameInstallType && sameName && sameVersion
        }).length > 0) {
            throw new AlreadyAddedWarning("The snapshot has already been added! Thus the operation is ignored!")
        }
        this.softwares.push(newSoftware)
    }

    /**
     * Returns the related software stored with the ID given by the user
     * @param {number} softwareIndex software key index in the array
     * @returns {SnapshotSoftware} Found software
     * @throws {NotFoundError} In case of not found element
     */
    getSoftware (softwareIndex: number): SnapshotSoftware {
        if (softwareIndex >= this.softwares.length) {
            throw new NotFoundError("The software ID exceeds the software array size!")
        }
        return this.softwares[softwareIndex]
    }
}
