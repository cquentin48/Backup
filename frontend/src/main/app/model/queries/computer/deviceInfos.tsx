import { createAsyncThunk } from "@reduxjs/toolkit";
import type BasicQueryParameters from "../basicQueryParameters";

import getDeviceInfos from "../../../../res/queries/computer_infos.graphql";
import gqlClient from "../client";
import Device from "../../device/device";
import SnapshotID from "../../device/snapshotId";
import { type ApolloQueryResult } from "@apollo/client";

interface QueryResult{
    deviceInfos: Device;
}

export const fetchDeviceInfos = createAsyncThunk(
    'device/deviceInfos',
    async (parameters: BasicQueryParameters, { rejectWithValue }) => {
        const result = (
            await gqlClient.execute_query(
                getDeviceInfos,
                parameters
            ) as ApolloQueryResult<QueryResult>
        )
        if (result.errors != null) {
            return rejectWithValue("The device wasn't found!")
        }

        const deviceData = result.data.deviceInfos;
        const rawSnapshots = deviceData.snapshots;
        const snapshots: SnapshotID[] = []
        rawSnapshots.forEach((element: any) => {
            snapshots.push(
                new SnapshotID(
                    element.key,
                    element.date,
                    element.operatingSystem
                )
            )
        });
        return new Device(
            deviceData.name,
            deviceData.processor,
            deviceData.cores,
            deviceData.memory,
            snapshots
        )
    }
);
