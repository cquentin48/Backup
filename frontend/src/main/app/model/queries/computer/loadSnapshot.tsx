import snapshotInfos from "../../../../res/queries/snapshot.graphql";
import gqlClient from "../client";
import type BasicQueryParameters from "../basicQueryParameters";
import { SnapshotData } from "../../snapshot/snapshotData";
import { createAsyncThunk } from "@reduxjs/toolkit";
import { type ApolloQueryResult } from "@apollo/client";

interface QueryResult {
    snapshotInfos: SnapshotData
}

export const fetchSnapshot = createAsyncThunk(
    'device/snapshotInfos',
    async (parameters: BasicQueryParameters, { rejectWithValue }) => {
        const result = (
            await gqlClient.execute_query(
                snapshotInfos,
                parameters
            ) as ApolloQueryResult<QueryResult>
        )
        try {
            if (result.errors != null) {
                return rejectWithValue("The snapshot you try to seek doesn't exist!")
            }
            const rawSnapshotData = result.data.snapshotInfos;
            const rawSoftwares = rawSnapshotData.versions;
            const snapshot = new SnapshotData();
            rawSoftwares.forEach((softwareRaw: any) => {
                const chosenVersion = softwareRaw.chosenVersion;
                const name = softwareRaw.name;
                const installType = softwareRaw.installType;
                snapshot.addSoftware(
                    chosenVersion,
                    name,
                    installType
                );
            });
            return snapshot;
        } catch (e) {
            return rejectWithValue("Selected snapshot doesn't exist!")
        }
    }
);
