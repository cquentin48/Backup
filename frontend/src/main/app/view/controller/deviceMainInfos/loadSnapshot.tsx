import ControllerAction from "../controllerActions";

import { filterManager } from "../../model/filters/FilterManager";
import snapshotData from "../../../../res/queries/snapshot.graphql";
import gqlClient from "../../../model/queries/client";
import { snapshotGQLData } from "../../../model/queries/computer/loadSnapshot";
import { type SnapshotData } from "../../../model/snapshot/snapshotData";
import { dataManager } from "../../../model/AppDataManager";

/**
 * "Adds new filter" controller action set in the device main information page.
 */
class LoadSnapshot extends ControllerAction {

    /**
     * Set new filter inside for the device main information view.
     * @param { string } inputs Inputs set by the user for the new filter.
     */
    performAction (inputs: string): void {
        const selectedSnapshotID = JSON.parse(inputs);
        const query = snapshotData;
        snapshotGQLData.computeQuery(
            gqlClient,
            query,
            {
                snapshotID: selectedSnapshotID
            }
        ).then((result: SnapshotData) => {
            dataManager.setElement("snapshot", result);

            const callBackMethod = [this.getObservable("MainInfosFrame"), this.getObservable("softwareInfosPieChart")]
            for (let i = 0; i < callBackMethod.length; i++) {
                callBackMethod[i](JSON.stringify(filterManager.getFilters()))
            }
        }).catch(e=>{
            const error = e as Error
            const stack = (error.stack as string)
            console.log(stack)
        })
    }
}

export const loadSnapshot = new LoadSnapshot();
