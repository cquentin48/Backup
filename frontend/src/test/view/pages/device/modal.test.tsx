import React from "react";

import { type DocumentNode, type FetchResult } from "@apollo/client";
import { type ResultFunction, MockedProvider } from "@apollo/client/testing";

import { type EnhancedStore, configureStore } from "@reduxjs/toolkit";

import { type RenderResult, render } from "@testing-library/react";

import { SnackbarProvider, useSnackbar } from "notistack";

import { useSelector, Provider, useDispatch } from "react-redux";

import snapshotReducer from "../../../../main/app/controller/deviceMainInfos/loadSnapshotSlice"
import filterReducer from "../../../../main/app/controller/deviceMainInfos/filterSlice"
import deviceReducer from "../../../../main/app/controller/deviceMainInfos/loadDeviceSlice"
import chatbotReducer from "../../../../main/app/controller/deviceMainInfos/chatbotSlice"
import { type AppState } from "../../../../main/app/controller/store";

import Device from "../../../../main/app/model/device/device";
import NotFoundError from "../../../../main/app/model/exception/errors/notFoundError";
import type Filter from "../../../../main/app/model/filters/Filter";
import gqlClient from "../../../../main/app/model/queries/client";
import { type DeviceInfosQueryResult } from "../../../../main/app/model/queries/computer/deviceInfos";
import { type LoadSnapshotQueryResult } from "../../../../main/app/model/queries/computer/loadSnapshot";
import { SnapshotData } from "../../../../main/app/model/snapshot/snapshotData";

import LoadingDeviceModal from "../../../../main/app/view/pages/computer/DeviceModal";

import FETCH_SNAPSHOT from '../../../../main/res/queries/snapshot.graphql';
import FETCH_DEVICE from '../../../../main/res/queries/computer_infos.graphql';

interface ApolloMockResult {
    request: {
        query: DocumentNode
    }
    result: FetchResult<LoadSnapshotQueryResult | DeviceInfosQueryResult> | ResultFunction<FetchResult<LoadSnapshotQueryResult | DeviceInfosQueryResult>, any> | undefined
}

jest.mock("notistack", () => {
    const actual = jest.requireActual("notistack");
    return {
        ...actual,
        useSnackbar: jest.fn()
    };
});

jest.mock("react-redux", () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

describe("Device loading modal test suite snapshot", () => {
    beforeEach(() => {
        gqlClient.get_query_client().query = initGraphQLMock()
    })
    afterEach(() => {
        jest.clearAllMocks()
    })

    /**
     * Mock the ``enqueueSnackbar`` function
     * @returns {jest.Mock} mocked ``enqueueSnackbar`` function
     */
    const initEnqueueSnackbarMock = (): jest.Mock => {
        const mockEnqueueSnackbar = jest.fn();
        (useSnackbar as jest.Mock).mockReturnValue({
            enqueueSnackbar: mockEnqueueSnackbar
        });

        return mockEnqueueSnackbar
    }

    /**
     * Init the ``useSelector`` mock for the unit test
     * @param {EnhancedStore<AppState>} store Mocked redux store
     */
    const initUseSelectorMock = (store: EnhancedStore<AppState>): void => {
        const mockedUseSelector = useSelector as jest.MockedFunction<typeof useSelector>;
        mockedUseSelector.mockImplementation((selector: (state: AppState) => unknown) => {
            const state = store.getState()

            const filterState = state.filter
            const deviceState = state.device
            const snapshotState = state.snapshot

            return selector(
                {
                    filter: {
                        filters: JSON.parse(JSON.stringify(filterState.filters)),
                        selectedFilteredIDS: filterState.selectedFilteredIDS,
                        filterError: {
                            message: filterState.filterError.message,
                            variant: filterState.filterError.variant
                        }
                    },
                    snapshot: {
                        snapshotError: snapshotState.snapshotError,
                        operationStatus: snapshotState.operationStatus,
                        snapshot: snapshotState.snapshot !== undefined ? JSON.parse(JSON.stringify(snapshotState.snapshot)) : undefined
                    },
                    device: {
                        device: deviceState.device !== undefined ? JSON.parse(JSON.stringify(deviceState.device)) : undefined,
                        deviceError: {
                            message: deviceState.deviceError.message,
                            variant: deviceState.deviceError.variant
                        },
                        deviceLoading: store.getState().device.deviceLoading
                    },
                    chatbot:{
                        conversationHeaders: [],
                        messages: []
                    }
                }
            )
        }
        )
    }

    const initApolloMock = (operationStatus: "success" | "error" | "loading" | "initial", snapshot: SnapshotData | undefined = undefined, device: Device | undefined = undefined): ApolloMockResult[] => {
        let snapshotResult: FetchResult<LoadSnapshotQueryResult> | ResultFunction<FetchResult<LoadSnapshotQueryResult>, any> | undefined;
        let deviceResult: FetchResult<DeviceInfosQueryResult> | ResultFunction<FetchResult<DeviceInfosQueryResult>, any> | undefined;

        if (operationStatus === "success") {
            if (snapshot === undefined && operationStatus !== "success") {
                throw new NotFoundError("Invalid operation : if the test type is a success, the snapshot must be defined!")
            }
            snapshotResult = {
                data: {
                    snapshotInfos: snapshot as SnapshotData
                }
            }
            deviceResult = {
                data: {
                    deviceInfos: device as Device
                }
            }

        } else if (operationStatus === "error") {
            const errorMessage = "Raised error message here!"
            snapshotResult = {
                errors: [
                    {
                        message: errorMessage
                    }
                ]
            }
            deviceResult = {
                errors: [
                    {
                        message: errorMessage
                    }
                ]
            }
        } else {
            snapshotResult = {
                data: {
                    snapshotInfos: new SnapshotData()
                }
            }
            deviceResult = {
                data: {
                    deviceInfos: new Device()
                }
            }
        }
        return [
            {
                request: {
                    query: FETCH_SNAPSHOT
                },
                result: snapshotResult
            },
            {
                request: {
                    query: FETCH_DEVICE
                },
                result: deviceResult
            }
        ]
    }

    /**
     * Render the SoftwaresOrigin component with the Apollo query and store mocks
     * @param {EnhancedStore} store Redux mocked store
     * @param {ApolloMockResult[]} apolloMocks Apollo GraphQL queries mocks result
     * @returns {NotFoundError} If the operation is marked as a success and no snapshot is provided.
     */
    const renderMockedComponent = (store: EnhancedStore, apolloMocks: ApolloMockResult[]): RenderResult => {
        return render(
            <Provider store={store}>
                <MockedProvider mocks={apolloMocks} addTypename={false}>
                    <SnackbarProvider>
                        <LoadingDeviceModal />
                    </SnackbarProvider>
                </MockedProvider>
            </Provider>
        )
    }

    /**
     * Initialise the test
     * @param {"success" | "error" | "loading" | "initial"} operationStatus Type of operation mocked for the unit test
     * @param {SnapshotData | undefined} snapshot Device snapshot used in the unit test
     * @param {Device |undefined} device device used for the mock
     * @param {Filter[]} filters Filters used in the unit test
     * @param {number[]} selectedFiltersIDs selected ids for the mock
     * @returns {EnhancedStore} Mocked store
     * @throws {Error} If the test stage is not in the list
     */
    const initStore = (operationStatus: "initial" | "loadingSnapshot" | "loadingDevice" | "success" | "error", snapshot: SnapshotData | undefined = undefined, device: Device | undefined = undefined, filters: Filter[] = [], selectedFiltersIDs: number[] = []): EnhancedStore<AppState> => {
        if (device === undefined && snapshot === undefined && operationStatus === "success") {
            throw new Error("The snapshot and the device must be defined if the loading snapshot data with a GraphQL query is successful!")
        }
        const parsedFilters = filters !== undefined ? JSON.parse(JSON.stringify(filters)) : undefined
        const parsedDevice = device !== undefined ? JSON.parse(JSON.stringify(device)) : undefined
        const parsedSnapshot = snapshot !== undefined ? JSON.parse(JSON.stringify(snapshot)) : undefined

        const preloadedState: AppState = {
            filter: {
                filters: parsedFilters,
                filterError: {
                    message: operationStatus === "error" ? "Snapshot : Error raised here!" : "",
                    variant: operationStatus === "error" ? "error" : undefined
                },
                selectedFilteredIDS: selectedFiltersIDs
            },
            device: {
                device: parsedDevice,
                deviceError: {
                    message: operationStatus === "error" ? "Snapshot : Error raised here!" : "",
                    variant: operationStatus === "error" ? "error" : undefined
                },
                deviceLoading: operationStatus === "initial" || operationStatus === "loadingDevice"
            },
            snapshot: {
                snapshot: parsedSnapshot,
                snapshotError: {
                    message: operationStatus === "error" ? "Device : Error raised here!" : "",
                    variant: operationStatus === "error" ? "error" : undefined
                },
                operationStatus: operationStatus === "loadingSnapshot" ? "loading" : (operationStatus !== "loadingDevice" ? operationStatus : "loading")
            },
            chatbot:{
                conversationHeaders: [],
                messages: []
            }
        }

        return configureStore({
            reducer: {
                device: deviceReducer,
                snapshot: snapshotReducer,
                filter: filterReducer,
                chatbot: chatbotReducer
            },
            preloadedState
        })
    }

    /**
     * Init graphql query mock for the unit tests
     * @returns {jest.Mock} Mocked graphql query function
     */
    const initGraphQLMock = (): jest.Mock => {
        const snapshotQueryOutput = {
            data: {
                snapshotInfos: {
                    versions: [
                        {
                            name: "My software",
                            chosenVersion: "type",
                            installType: "1.0"
                        }
                    ],
                    repositories: []
                }
            }
        }
        return jest.fn().mockReturnValue(snapshotQueryOutput)
    }

    test("Before loading", async () => {
        // Before
        const mockedDispatch = jest.fn();
        (useDispatch as jest.MockedFunction<typeof useDispatch>).mockImplementation(() => {
            return mockedDispatch
        });
        initEnqueueSnackbarMock()
        expect(true).toBe(true)

        // Given
        const store = initStore("initial")

        initUseSelectorMock(store)

        const apolloMocks = initApolloMock("initial")

        // Acts
        const { asFragment } = renderMockedComponent(store, apolloMocks)

        // Asserts
        expect(asFragment()).toMatchSnapshot()
    })

    test("Loading device", async () => {
        // Before
        const mockedDispatch = jest.fn();
        (useDispatch as jest.MockedFunction<typeof useDispatch>).mockImplementation(() => {
            return mockedDispatch
        });
        initEnqueueSnackbarMock()
        expect(true).toBe(true)

        // Given
        const store = initStore("loadingDevice")

        initUseSelectorMock(store)

        const apolloMocks = initApolloMock("loading")

        // Acts
        const { asFragment } = renderMockedComponent(store, apolloMocks)

        // Asserts
        expect(asFragment()).toMatchSnapshot()
    })

    test("Loading snapshot", async () => {
        // Before
        const mockedDispatch = jest.fn();
        (useDispatch as jest.MockedFunction<typeof useDispatch>).mockImplementation(() => {
            return mockedDispatch
        });
        initEnqueueSnackbarMock()
        expect(true).toBe(true)

        // Given
        const store = initStore("loadingSnapshot")

        initUseSelectorMock(store)

        const apolloMocks = initApolloMock("loading")

        // Acts
        const { asFragment } = renderMockedComponent(store, apolloMocks)

        // Asserts
        expect(asFragment()).toMatchSnapshot()
    })

    test("After loading", async () => {
        // Before
        const mockedDispatch = jest.fn();
        (useDispatch as jest.MockedFunction<typeof useDispatch>).mockImplementation(() => {
            return mockedDispatch
        });
        initEnqueueSnackbarMock()
        expect(true).toBe(true)

        const snapshot = new SnapshotData()
        const device = new Device()

        // Given
        const store = initStore("success", snapshot, device)

        initUseSelectorMock(store)

        const apolloMocks = initApolloMock("success")

        // Acts
        const { asFragment } = renderMockedComponent(store, apolloMocks)

        // Asserts
        expect(asFragment()).toMatchSnapshot()
    })
})
