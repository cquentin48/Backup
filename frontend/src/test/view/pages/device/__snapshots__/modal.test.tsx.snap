// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device loading modal test suite snapshot After loading 1`] = `<DocumentFragment />`;

exports[`Device loading modal test suite snapshot Before loading 1`] = `<DocumentFragment />`;

exports[`Device loading modal test suite snapshot Loading device 1`] = `<DocumentFragment />`;

exports[`Device loading modal test suite snapshot Loading snapshot 1`] = `<DocumentFragment />`;
